### DISCUSSION TITLE
Lab 02.01

### DISCUSSION THREAD

**Discussion by danielsalazarg** [rramosp/ai4eng.v1#133] [UDEA Students]
Hola
He intentado realizar este laboratorio pero no encuentro la manera de que me arroje el resultado, ¿que debería hacer?
![02 01](https://user-images.githubusercontent.com/106793308/171780957-a0835fd9-1455-4d78-91e8-5205741421e8.png)

**Comment by Felipe-RA** [comment]
Saludos danielsalazar

El primer paso cuando tienes un error es buscarlo en internet. De ahí es empezar a tirar del hilo y llegar a la solución.

En este caso, ¿qué significa `int object is not subscriptable`?
¿Comprendes los requisitos para que `n == n[::-1]` funcione?
¿Sabes cómo funciona esa notación?

Responderte a ti mismo estas preguntas hace parte del camino para solucionar el problema general.

Y si después de intentartelas responder no logras solucionar, aquí estamos para echarte una mano.

Me comentas cuando hagas lo de arriba!

**Comment by danielsalazarg** [comment]
Buenas tardes
Si publique en el foro mi problema fué por que no pude encontrar nada que
me ayudara, n==n[::-1] es para que python lo lea al revés y poder comparar
si son iguales, leí que tal vez cambiando n de int a str funcionaba pero no
lo conseguí, Ayer me gaste casi todo el día buscando una solución.

El jue., 2 de junio de 2022 11:08 p. m., Felipe-RA ***@***.***>
escribió:

> Saludos danielsalazar
>
> El primer paso cuando tienes un error es buscarlo en internet. De ahí es
> empezar a tirar del hilo y llegar a la solución.
>
> En este caso, ¿qué significa int object is not subscriptable?
> ¿Comprendes los requisitos para que n == n[::-1] funcione?
> ¿Sabes cómo funciona esa notación?
>
> Responderte a ti mismo estas preguntas hace parte del camino para
> solucionar el problema general.
>
> Y si después de intentartelas responder no logras solucionar, aquí estamos
> para echarte una mano.
>
> Me comentas cuando hagas lo de arriba!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/133#discussioncomment-2875639>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZOYSXAREGD2NT54QH5AVYLVNGALNANCNFSM5XXFHJMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
"La información aquí contenida es para uso exclusivo de la persona o 
entidad de destino. Está estrictamente prohibida su utilización, copia, 
descarga, distribución, modificación y/o reproducción total o parcial, sin 
el permiso expreso de Universidad de Antioquia, pues su contenido puede ser 
de carácter confidencial y/o contener material privilegiado. Si usted 
recibió esta información por error, por favor contacte en forma inmediata a 
quien la envió y borre este material de su computador. Universidad de 
Antioquia no es responsable por la información contenida en esta 
comunicación, el directo responsable es quien la firma o el autor de la 
misma."

-- 
UdeA

**Reply by Felipe-RA** [reply]
Hola daniel

Como podrás comprender no tengo forma de saber qué hiciste y qué no hiciste antes de preguntar en el foro (más allá de lo que incluyes en tu pregunta, que en este caso no fue indicador de que hayas echo  introspección en tu problema)

Para aprender algo nuevo de este post para quienes puedan leerlo en el futuro:

Que `n[::-1]` sea para que "lo lea al revés" no es correcto. Lo que hace esta notación es recorrer un contenedor indexable de principio a fin en pasos de -1 `[start:stop:step]`

En donde, en este caso, se recorre en su totalidad pero en pasos negativos (uno a uno desde el final hasta el comienzo)

Ahora, los `subscriptable objects` son cualquier objeto que sea un contenedor y que implemente el método `__get_item__()`. Son estos los objetos que permiten usar la notacion `n[ ]`

Un ejemplo de estos objetos serían las listas, los **strings**, las tuplas, etc.

Entonces, cuando te dicen que un objeto int no es subscribible, te dicen que no puedes usar esa notación sobre ellos.

Ahora, con ese conocimiento puedes solucionar el problema de algunas maneras:

- Convertir el int a otro objeto que sea subscribible y se encargue de separarte los dígitos.
- Separarlos tu mismo con operaciones matemáticas.
